migrator/zenodo_rdm_migrator/extract/kafka.py

```diff
@@ -7,33 +7,291 @@
 
 """Kafka extraction classes."""
 
-from invenio_rdm_migrator.extract import Extract
-from kafka import KafkaConsumer
+import json
+from collections import Counter
+from datetime import datetime
+from pathlib import Path
+
+from invenio_rdm_migrator.extract import Extract, Tx
+from invenio_rdm_migrator.logging import Logger
+from kafka import KafkaConsumer, TopicPartition
+from sortedcontainers import SortedDict, SortedList
+
+
+class TxState:
```

Comment:
Should this be moved along with `Tx`? or otherwise be called `KafkaTxState`, if it is very tied to kafka behaviour?

Reply:
Yeah, it's more like a "pending transaction container" that helps build it and easily check if a transaction is "complete". Originally I called it `PendingTx` (but felt wrong to check for `pending_tx.complete`)... It's only used internally, so either prefixing with `KafkaTxState` or making it private (e.g. `_TxState`) would also work.

Reply:
Renamed in the latest commit to `_TxState`.